### PR TITLE
Run multi-node tests on fresh GKE clusters

### PR DESCRIPTION
**Pull request by andreaTP**
Notes:

- I have integrated the source code of `sbt-multi-jvm` to change it to rely exclusively on `kubectl` commands.
- `akka-cluster-metrics` tests are already passing and most of the rest, but a few tests will require further attention.
- Concurrent executions are not a problem since the cluster name is generated based on the unique Job ID in CI.
- The overhead of creating/shutting down the entire cluster is pretty low and is about 10 minutes.

We used to run those tests on PR validation, but I'm unsure if that's still the case, I would need a little additional guidance to get this PR ready to be merged.

### CODE REVIEW

**Comment by andreaTP** [comment]
example green CI:
https://github.com/andreaTP/akka/runs/3413411100

**Review by marcospereira — COMMENTED** [review]
> Concurrent executions are not a problem since the cluster name is generated based on the unique Job ID in CI.

But we can still run into account's limitations, right?

---

I've made some comments.

Overall, I think that copying and adapting the code from sbt-multi-jvm is strange. Can we extend the plugin to support this new use case instead?

**Comment by patriknw** [comment]
> We used to run those tests on PR validation

No, we didn't run multi-node nor multi-jvm in PR validation, and we don't want that now either because it would take too long time.

**Review by patriknw — COMMENTED** [review]
looking good

File: .github/workflows/multi-node.yml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+          to: akka.official@gmail.com
+          from: Akka CI (GHActions)
+          body: |
+            Multi node test of ${{github.repository}} failed!
+            https://github.com/${{github.repository}}/actions/runs/${{github.run_id}}
+      - name: Cleanup the environment
+        if: ${{ always() }}
+        shell: bash {0}
+        run: |
+          gcloud container clusters delete "akka-multi-node-${GITHUB_RUN_ID}" --quiet
```
[the remark this answers is not in the document]
Comment (andreaTP) [reply .github/workflows/multi-node.yml:93]: `--quiet` actually means "non-interactive", is needed to run in CI

File: kubernetes/create-cluster-gke.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    exit 1
+fi
+
+CLUSTER_NAME=$1
+CLUSTER_VERSION=$2
+
+if [ -z "$CLUSTER_VERSION" ]
+  then
+    # https://cloud.google.com/kubernetes-engine/versioning-and-upgrades#versions_available_for_new_cluster_masters
+    CLUSTER_VERSION=$(gcloud container get-server-config --format json | jq -r .defaultClusterVersion)
```
[the remark this answers is not in the document]
Comment (andreaTP) [reply kubernetes/create-cluster-gke.sh:56]: ok

**Comment by andreaTP** [comment]
@marcospereira :

> But we can still run into account's limitations, right?

yes, but it's a very high limit compared to the fact we will run this job only nightly.

> Can we extend the plugin to support this new use case instead?

Yes, eventually, the plugin requires quite some maintenance activity at the moment (e.g. support for sbt 0.13, publishing to Bintray ...)

File: .github/workflows/multi-node.yml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      - name: Install Kubectl
+        run: |
+          sudo snap install kubectl --classic
+      # - name: Setup Minikube
+      #   uses: manusa/actions-setup-minikube@v2.4.2
+      #   with:
+      #     minikube version: 'v1.21.0'
+      #     kubernetes version: 'v1.21.0'
+      #     driver: docker
+      # Latest tagged version uses a deprecated github actions feature that is going to be removed on 16th nov
```
[the remark this answers is not in the document]
Comment (andreaTP) [reply .github/workflows/multi-node.yml]: updated

File: .github/workflows/multi-node.yml [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+      - uses: GoogleCloudPlatform/github-actions/setup-gcloud@15dc2ebff158fa77e9b28e43515bae5f990345b4
+        with:
+          version: '290.0.1'
```
[the remark this answers is not in the document]
Comment (andreaTP) [reply .github/workflows/multi-node.yml]: removed the fixed version

**Review by patriknw — COMMENTED** [review]
excellent, great work

**Review by marcospereira — COMMENTED** [review]
Some minor suggestions and questions.

Great work @andreaTP!

**Review by patriknw — APPROVED** [review]
LGTM

**Review by marcospereira — COMMENTED** [review]
LGTM.

Great work, @andreaTP!

**Pull request merged**